jumanji/environments/packing/bin_pack/env.py

```diff
@@ -864,6 +822,112 @@ def observation_spec(self) -> specs.Spec[Observation]:
+                (max_num_items,), float, -1.0, 1.0, "value"
+            )
+        else:
+            items_dict["value"] = specs.BoundedArray(
+                (max_num_items,), float, -jnp.inf, jnp.inf, "value"
+            )
+        return items_dict
+
+    def _items_dict_for_non_valued_items(
+        self, max_num_items: int, max_dim: int
+    ) -> Dict:
+        """Set the items_dict specs to the correct bounded array for non valued items depending
+        on whether dimensions are to be normalized or not.
+
+        Args:
+            max_num_items: the maximum number of items that can be in an instance.
+            max_dim: The maximum dimension in this given instance.
+
+        Returns:
+            A dictionary with string keys of the item features and specs BoundedArray as values.
+        """
+        if self.normalize_dimensions:
+            return {
+                f"{axis}": specs.BoundedArray((max_num_items,), float, 0.0, 1.0, axis)
+                for axis in ["x_len", "y_len", "z_len"]
+            }
+        return {
+            f"{axis}": specs.BoundedArray((max_num_items,), jnp.int32, 0, max_dim, axis)
+            for axis in ["x_len", "y_len", "z_len"]
+        }
+
+    def _items_dict_for_rotated_items(self, max_num_items: int, max_dim: int) -> Dict:
+        """Set the items_dict specs to the correct bounded array for items depending
+        on whether features are to be normalized or not.
+
+        Args:
+            max_num_items: the maximum number of items that can be in an instance.
+            max_dim: The maximum dimension in this given instance.
+
+        Returns:
+            A dictionary with string keys of the item features and specs BoundedArray as values.
+        """
+        if self.normalize_dimensions:
+            items_dict = {
+                f"{axis}": specs.BoundedArray(
+                    (6 * max_num_items,), float, 0.0, 1.0, axis
+                )
+                for axis in ["x_len", "y_len", "z_len"]
+            }
+
+        else:
+            items_dict = {
+                f"{axis}": specs.BoundedArray(
+                    (6 * max_num_items,), jnp.int32, 0, max_dim, axis
+                )
+                for axis in ["x_len", "y_len", "z_len"]
+            }
+        return items_dict
+
+    def _items_dict_for_rotated_valued_items(
+        self, max_num_items: int, max_dim: int
+    ) -> Dict:
+        """Set the items_dict specs to the correct bounded array for valued items depending
+        on whether features are to be normalized or not.
+
+        Args:
+            max_num_items: the maximum number of items that can be in an instance.
+            max_dim: The maximum dimension in this given instance.
+
+        Returns:
+            A dictionary with string keys of the item features and specs BoundedArray as values.
+        """
+        items_dict = self._items_dict_for_valued_items(max_num_items, max_dim)
```

Comment:
Would it be easier for this line to be replaced by _items_dict_for_rotated_items and then add the valued bit on top (then the rotated items bit below can be removed. At the moment it looks like everything in items_dict is being overwritten by what is below.

Reply:
You're right. I added it.